### ISSUE TITLE
2.9.0 hanging at boot and not applying AFWall rules

### ISSUE THREAD

**Issue by F-i-f** [ukanth/afwall#623]
System information (software):
OmniROM 6.0.1 20161123
SuperSU 2.78 SR4
Xposed v87
AFWall+ 2.9.0 (Donate)

Problem seen on (hardware):
Nexus 5 (hammerhead)
Nexus 4X (bullhead)
Nexus 10 (manta)

Symptoms:
1. Reboot system
2. Wait a long while for the boot to fully complete
3. Launch root shell (adb/Terminal)
4. Check that the iptables rules have not been applied. (iptables -L doesn't show the afwall chains)
5. Launch the AFWall app.
6. Using the three dots menu, start "Apply"
7. AFWall displays the popup "Applying iptables rules"
8. AFWall stays stuck there forever (killed it after 10 minutes)

Expected behavior:
At step 4, I'd expect to see the afwall iptables rules.
At step 8, AFWall should complete the iptables set-up and allow interacting with the app again.

Additional notes:
The problem seems to happen only randomly.  Out of 7 boots, 3 of them ended up with a stuck AFWall.

Logcat:
This logcat may have missed early boot.
```
11-27 10:57:28.594 10264 16232 D AFWall  : Applying rules on connectivity change
11-27 10:57:28.627 10264 16232 D AFWall  : isWifiApEnabled is false
11-27 10:57:29.019 10264 16232 I AFWall  : Now assuming 3G connection (non-tethered)
11-27 10:57:29.020 10264 16232 I AFWall  : Using fastApply
11-27 10:57:29.268 10264 16232 I AFWall  : Setting OUTPUT chain to DROP
11-27 10:57:29.269 10264 16232 I AFWall  : Setting OUTPUT chain to ACCEPT
11-27 10:57:29.398 10264 16232 I AFWall  : Setting OUTPUT chain to DROP
11-27 10:57:29.398 10264 16232 I AFWall  : Setting OUTPUT chain to ACCEPT
11-27 10:57:29.406 10264 10264 D AFWall  : Received request to kill logservice
11-27 10:57:29.407 10264 10264 I AFWall  : [libsuperuser] [SU%] START
11-27 10:57:29.428 10264 16232 D AFWall  : Applying rules on connectivity change
11-27 10:57:29.434 10264 16232 D AFWall  : isWifiApEnabled is false
11-27 10:57:29.480 10264 10264 I AFWall  : Cleaning up previous uid: 15655
11-27 10:57:29.499 10264 10264 I AFWall  : Starting Log Service: echo PID=$$ & while true; do dmesg -c ; sleep 1 ; done for LogTarget: LOG
11-27 10:57:29.500 10264 10264 I AFWall  : rootSession is not Null
11-27 10:57:29.500 10264 10264 I AFWall  : [libsuperuser] [SU%] START
11-27 10:57:29.594 10264 16232 D AFWall  : CONNECTIVITY_CHANGE: interface state has not changed, ignoring
11-27 10:57:29.607 10264 10264 D AFWall  : Received request to kill logservice
11-27 10:57:29.614 10264 10264 I AFWall  : Starting Log Service: echo PID=$$ & while true; do dmesg -c ; sleep 1 ; done for LogTarget: LOG
11-27 10:57:29.614 10264 10264 I AFWall  : rootSession is not Null
11-27 10:57:29.615 10264 10264 I AFWall  : [libsuperuser] [SU%] START
11-27 10:57:37.485 10264 16621 D AFWall  : Applying rules on connectivity change
11-27 10:57:37.512 10264 16621 D AFWall  : isWifiApEnabled is false
11-27 10:57:37.778 10264 16621 I AFWall  : Now assuming wifi connection
11-27 10:57:37.778 10264 16621 I AFWall  : IPv4 LAN netmask on wlan0: 172.17.2.5/27
11-27 10:57:37.778 10264 16621 I AFWall  : IPv6 LAN netmask on wlan0: fe80::66bc:cff:fe51:35fb/64
11-27 10:57:37.779 10264 16621 I AFWall  : Using fastApply
11-27 10:57:37.985 10264 16621 I AFWall  : Setting OUTPUT chain to DROP
11-27 10:57:37.986 10264 16621 I AFWall  : Setting OUTPUT chain to ACCEPT
11-27 10:57:38.115 10264 16621 I AFWall  : Setting OUTPUT chain to DROP
11-27 10:57:38.116 10264 16621 I AFWall  : Setting OUTPUT chain to ACCEPT
11-27 10:57:38.124 10264 10264 D AFWall  : Received request to kill logservice
11-27 10:57:38.125 10264 10264 I AFWall  : [libsuperuser] [SU%] START
11-27 10:57:38.144 10264 16621 D AFWall  : Applying rules on connectivity change
11-27 10:57:38.150 10264 16621 D AFWall  : isWifiApEnabled is false
11-27 10:57:38.183 10264 10264 I AFWall  : Cleaning up previous uid: 16338
11-27 10:57:38.201 10264 10264 I AFWall  : Starting Log Service: echo PID=$$ & while true; do dmesg -c ; sleep 1 ; done for LogTarget: LOG
11-27 10:57:38.201 10264 10264 I AFWall  : rootSession is not Null
11-27 10:57:38.202 10264 10264 I AFWall  : [libsuperuser] [SU%] START
11-27 10:57:38.245 10264 16621 D AFWall  : CONNECTIVITY_CHANGE: interface state has not changed, ignoring
11-27 10:57:38.258 10264 10264 D AFWall  : Received request to kill logservice
11-27 10:57:38.260 10264 10264 I AFWall  : Starting Log Service: echo PID=$$ & while true; do dmesg -c ; sleep 1 ; done for LogTarget: LOG
11-27 10:57:38.260 10264 10264 I AFWall  : rootSession is not Null
11-27 10:57:38.261 10264 10264 I AFWall  : [libsuperuser] [SU%] START
```
At this point, the iptables rules are unset:
`iptables -L OUTPUT` does not show an `afwall` rule.
I then try to apply the rules through the app's Apply button:
```
11-27 11:01:50.279 10264 10264 I AFWall  : applySavedIptablesRules invoked
11-27 11:01:50.388 10264 10264 I AFWall  : Setting OUTPUT to Drop
11-27 11:01:50.390 10264 10264 I AFWall  : Setting OUTPUT to Accept
11-27 11:01:50.547 10264 10264 I AFWall  : Setting OUTPUT to Drop
11-27 11:01:50.550 10264 10264 I AFWall  : Setting OUTPUT to Accept
```
And then it's hung.
AFWall keeps emitting messages though:
```
11-27 11:03:13.915 10264 21896 D AFWall  : Applying rules on connectivity change
11-27 11:03:13.922 10264 21896 D AFWall  : isWifiApEnabled is false
11-27 11:03:13.956 10264 21896 I AFWall  : Now assuming 3G connection (non-tethered)
11-27 11:03:13.957 10264 21896 I AFWall  : Using fastApply
11-27 11:03:14.469 10264 21896 I AFWall  : Setting OUTPUT chain to DROP
11-27 11:03:14.469 10264 21896 I AFWall  : Setting OUTPUT chain to ACCEPT
11-27 11:03:14.732 10264 21896 I AFWall  : Setting OUTPUT chain to DROP
11-27 11:03:14.732 10264 21896 I AFWall  : Setting OUTPUT chain to ACCEPT
11-27 11:03:14.745 10264 10264 D AFWall  : Received request to kill logservice
11-27 11:03:14.746 10264 10264 I AFWall  : [libsuperuser] [SU%] START
11-27 11:03:14.779 10264 10264 I AFWall  : Cleaning up previous uid: 16702
11-27 11:03:14.801 10264 10264 I AFWall  : Starting Log Service: echo PID=$$ & while true; do dmesg -c ; sleep 1 ; done for LogTarget: LOG
11-27 11:03:14.802 10264 10264 I AFWall  : rootSession is not Null
11-27 11:03:14.803 10264 10264 I AFWall  : [libsuperuser] [SU%] START
11-27 11:03:16.307 10264 22086 D AFWall  : Applying rules on connectivity change
11-27 11:03:16.310 10264 22086 D AFWall  : isWifiApEnabled is false
11-27 11:03:16.339 10264 22086 I AFWall  : Now assuming 3G connection (non-tethered)
11-27 11:03:16.347 10264 22086 I AFWall  : IPv6 LAN netmask on wlan0: fe80::66bc:cff:fe51:35fb/64
11-27 11:03:16.347 10264 22086 I AFWall  : Using fastApply
11-27 11:03:17.030 10264 22086 I AFWall  : Setting OUTPUT chain to DROP
11-27 11:03:17.030 10264 22086 I AFWall  : Setting OUTPUT chain to ACCEPT
11-27 11:03:17.429 10264 22086 I AFWall  : Setting OUTPUT chain to DROP
11-27 11:03:17.429 10264 22086 I AFWall  : Setting OUTPUT chain to ACCEPT
11-27 11:03:17.457 10264 10264 D AFWall  : Received request to kill logservice
11-27 11:03:17.457 10264 10264 I AFWall  : [libsuperuser] [SU%] START
11-27 11:03:17.488 10264 10264 I AFWall  : Cleaning up previous uid: 21988
11-27 11:03:17.510 10264 10264 I AFWall  : Starting Log Service: echo PID=$$ & while true; do dmesg -c ; sleep 1 ; done for LogTarget: LOG
11-27 11:03:17.510 10264 10264 I AFWall  : rootSession is not Null
11-27 11:03:17.511 10264 10264 I AFWall  : [libsuperuser] [SU%] START
11-27 11:03:18.657 10264 22342 D AFWall  : Applying rules on connectivity change
11-27 11:03:18.661 10264 22342 D AFWall  : isWifiApEnabled is false
11-27 11:03:18.691 10264 22342 I AFWall  : Now assuming wifi connection
11-27 11:03:18.696 10264 22342 I AFWall  : IPv4 LAN netmask on wlan0: 172.17.2.5/27
11-27 11:03:18.697 10264 22342 I AFWall  : IPv6 LAN netmask on wlan0: fe80::66bc:cff:fe51:35fb/64
11-27 11:03:18.697 10264 22342 I AFWall  : Using fastApply
11-27 11:03:19.211 10264 22342 I AFWall  : Setting OUTPUT chain to DROP
11-27 11:03:19.211 10264 22342 I AFWall  : Setting OUTPUT chain to ACCEPT
11-27 11:03:19.433 10264 22342 I AFWall  : Setting OUTPUT chain to DROP
11-27 11:03:19.434 10264 22342 I AFWall  : Setting OUTPUT chain to ACCEPT
11-27 11:03:19.444 10264 22342 D AFWall  : Applying rules on connectivity change
11-27 11:03:19.448 10264 22342 D AFWall  : isWifiApEnabled is false
11-27 11:03:19.457 10264 10264 D AFWall  : Received request to kill logservice
11-27 11:03:19.458 10264 10264 I AFWall  : [libsuperuser] [SU%] START
11-27 11:03:19.492 10264 10264 I AFWall  : Cleaning up previous uid: 22200
11-27 11:03:19.505 10264 10264 I AFWall  : Starting Log Service: echo PID=$$ & while true; do dmesg -c ; sleep 1 ; done for LogTarget: LOG
11-27 11:03:19.506 10264 10264 I AFWall  : rootSession is not Null
11-27 11:03:19.506 10264 10264 I AFWall  : [libsuperuser] [SU%] START
11-27 11:03:19.523 10264 22342 D AFWall  : CONNECTIVITY_CHANGE: interface state has not changed, ignoring
11-27 11:03:19.536 10264 10264 D AFWall  : Received request to kill logservice
11-27 11:03:19.538 10264 10264 I AFWall  : Starting Log Service: echo PID=$$ & while true; do dmesg -c ; sleep 1 ; done for LogTarget: LOG
11-27 11:03:19.538 10264 10264 I AFWall  : rootSession is not Null
11-27 11:03:19.539 10264 10264 I AFWall  : [libsuperuser] [SU%] START
```
At this point, AFWall was still showing "Applyting iptables rules", iptables were still unset.

**Comment by ukanth** [comment]
This is addressed in 2.9.1, Please test it and let me know.

**Comment by F-i-f** [comment]
I have no more issues with 2.9.1, tested against many reboots and many devices :-)
